en/news/_posts/2018-03-28-unintentional-file-and-directory-creation-with-directory-traversal-cve-2018-6914.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: news_post
+title: "CVE-2018-6914: Unintentional file and directory creation with directory traversal in tempfile and tmpdir"
+author: "usa"
+date: 2018-03-28 14:00:00 +0000
+tags: security
+lang: en
+---
+
+There is an unintentional directory creation vulnerability in tmpdir library bundled with Ruby.
+And there is also an unintentional file creation vulnerability in tempfile library bundled with Ruby, because it uses tmpdir internally.
+This vulnerability has been assigned the CVE identifier [CVE-2018-6914](http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2018-6914).
+
+## Details
+
+`Dir.mktmpdir` method introduced by tmpdir library accepts the prefix and the suffix of the directory which is created as the first parameter.
+The prefix can contain relative directory specifiers `"../"`, so this method can be used to target any directory.
+So, if a script accepts an external input as the prefix, and the targeted directory has inappropriate permissions or the ruby process has inappropriate privileges, the attacker can create a directory or a file at any directory.
+
+All users running an affected release should upgrade immediately.
+
+## Affected Versions
+
+* Ruby 2.2 series: 2.2.9 and earlier
+* Ruby 2.3 series: 2.3.6 and earlier
+* Ruby 2.4 series: 2.4.3 and earlier
+* Ruby 2.5 series: 2.5.0 and earlier
```

Comment:
@unak Is this `and earlier` intentional?
It is not match with japanese one.

Reply:
Oh, I didn't take care of the nuance.
Of course, Ruby 2.5 series had only one release 2.5.0, so we don't have to mention about "earlier".
But, about thinking preview versions, "earlier" might have some meaning.
Conclusion: you can choose to say or not to say "earlier" as you like :)

Reply:
Thanks for your reply :)